Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp

```diff
@@ -107,15 +107,37 @@ bool RenderSVGResourceMasker::applyResource(RenderElement& renderer, const Rende
 
 bool RenderSVGResourceMasker::drawContentIntoMaskImage(MaskerData* maskerData, const DestinationColorSpace& colorSpace, RenderObject* object)
 {
-    GraphicsContext& maskImageContext = maskerData->maskImage->context();
+    auto& maskImageContext = maskerData->maskImage->context();
+    auto objectBoundingBox = object->objectBoundingBox();
+
+    if (!drawContentIntoContext(maskImageContext, objectBoundingBox))
+        return false;
+
+#if !USE(CG)
+    maskerData->maskImage->transformToColorSpace(colorSpace);
+#else
+    UNUSED_PARAM(colorSpace);
+#endif
+
+    // Create the luminance mask.
+    if (style().svgStyle().maskType() == MaskType::Luminance)
```

Comment:
Is this the right place to do this, if there's a list of mask images (only some of which might be SVG references)?